src/reflect/scala/reflect/internal/tpe/TypeComparers.scala

```diff
@@ -351,6 +351,7 @@ trait TypeComparers {
   // @assume tp1.isHigherKinded || tp2.isHigherKinded
   def isHKSubType(tp1: Type, tp2: Type, depth: Depth): Boolean = {
     def isSub(ntp1: Type, ntp2: Type) = (ntp1.withoutAnnotations, ntp2.withoutAnnotations) match {
+      case (TypeRef(_, AnyClass, _), PolyType(_, _))                        => true                     // Any is kind-polymorphic
```

Comment:
Those answers look right to me. `List[Any]` is a proper type to which `Any` does not conform. The only proper type `Any` conforms to is `Any`. But the type constructor `List` is unapplied - that's why Any "conforms" to it, because all it's really doing is allowing it to survive this check.